### DISCUSSION TITLE
Downloaded song not showing in internal storage.

### DISCUSSION THREAD

**Discussion by max5354** [z-huang/music#108] [Q&A] [answered]
I download 1 song. But it only visible in "songs" tab. Why that songs not showing in my internal storage? Is it bug?

**Comment by z-huang** [comment, marked as answer]
The songs are downloaded to the app-specific storage, under /Android/data/com.zionhuang.music/file/media. It's designed not to be exposed to the public storage. I consider using SAF for users to access the files, but it's not implemented.

**Comment by FinallyCanuck** [comment]
I understand the answer, but how do you play the stored tracks. Is there an offline/airplane mode that allows you to listen to the stored tracks?

**Reply by z-huang** [reply]
If the song is downloaded, we don't need to trigger a network request to YouTube. We play the downloaded file directly.

**Reply by FinallyCanuck** [reply]
Thank you for the clarification. I have another app that, even if the song is stored locally, it will pull from the network if there's a connection.

**Reply by z-huang** [reply]
However, there's a bug that may cause downloaded song unable to play. See #329.
